### DISCUSSION TITLE
Welcome to AbsBox Discussions!

### DISCUSSION THREAD

**Discussion by yellowbean** [yellowbean/AbsBox#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions when you are using `absbox`
  * Enhancement , Bug reporting, or deal modelling request and deal sample request.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by quantum-ml-geek** [comment]
Can you recommend any good books that goes over the various parts of the ABS? Specifically Triggers. Outstanding repository!

**Reply by yellowbean** [reply]
<Introduction to Structured Finance> from Fabozzi is always a good start overall.
Triggers is a credit enhance mechanism which varies deal by deal, thus no book covers in detail, I would recommend reading prospectus of deal or offering circular / rating reports .

**Reply by quantum-ml-geek** [reply]
Thanks for the reply.  I have a good chunk of the Fabozzi books as well as several other Fixed Income books.  None of them really mentions anything about Triggers.  I guess going to the prospectus is always an option.

**Reply by yellowbean** [reply]
this is a abstraction of `Trigger` ( opinionated based )
https://absbox-doc.readthedocs.io/en/latest/modeling.html#trigger

its about when `bad` things happed, then the deal cashflow distribution would be changed to favor of senior bond holder.

`bad` things could be:
* deteriorated pool performance ( pool default rate/ delinquency rate/ loss rate  breach a threshold or breach a time series based threshold
* a servicer go bankruptcy 
* a bond rating went down
* or issuer has a lower rating
* a bond is outstanding too long without being paid off 
* a deal last too long without clean up call or paid off totally
